src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSyncController.scala

```diff
@@ -0,0 +1,193 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.actor._
+import io.iohk.ethereum.blockchain.sync.SyncController.StartRegularSync
+import io.iohk.ethereum.domain.BlockHeader
+import io.iohk.ethereum.network.PeerActor.Status.{Chain, Handshaked}
+import io.iohk.ethereum.network.PeerActor._
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.NewBlock
+import io.iohk.ethereum.network.p2p.messages.PV62._
+import io.iohk.ethereum.utils.Config
+import org.spongycastle.util.encoders.Hex
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.util.Try
+
+trait RegularSyncController {
+  self: SyncController =>
+
+  private var headersQueue: Seq[BlockHeader] = Seq.empty
+  private var resolvingBranch = false
+  private var broadcasting = false
+  private var timeout: Option[Cancellable] = None
+
+  import Config.FastSync._
+
+  def regularSync(): Receive = {
+
+    case StartRegularSync =>
+      askForHeaders()
+
+    case ResumeRegularSync =>
+      askForHeaders()
+
+    case MessageReceived(m: BlockHeaders) if !resolvingBranch =>
+      cancelTimeout()
+      handleDownload(m)
+
+    case MessageReceived(m: BlockHeaders) if resolvingBranch =>
+      cancelTimeout()
+      handleBlockBranchResolution(m)
+
+    case MessageReceived(m: BlockBodies) =>
+      cancelTimeout()
+      handleBlockBodies(m)
+
+    case m: BroadcastBlocks if broadcasting =>
+      peersToDownloadFrom.keys.foreach(_ ! m)
+
+    case ResolveBranch(peer) =>
+      requestHeadersForNewBranch(peer)
+
+    case PeerTimeOut(peer) =>
+      blacklist(peer, blacklistDuration)
+      askForHeaders()
```

Comment:
Should we set `headersQueue` to `Nil` here?
What happens if this timeout happens while resolving the branch, should we also set `resolvingBranch` to `false`?